packages/workbox-build/src/lib/utils/filter-files.js

```diff
@@ -19,23 +61,32 @@ module.exports = (fileDetails, options) => {
     return true;
   });
 
-  // Convert to manifest format
-  return filteredFileDetails.map((fileDetails) => {
-    let url = fileDetails.file.replace(path.sep, '/');
-
-    // Modify URL Prefix
-    if (options && options.modifyUrlPrefix) {
-      url = modifyUrlPrefixes(url, options.modifyUrlPrefix);
-    }
-
-    if (options.dontCacheBustUrlsMatching &&
-      url.match(options.dontCacheBustUrlsMatching)) {
-      return url;
-    }
-
+  // Take the array of fileDetail objects and convert it into an array of
+  // {url, revision} objects, with path.sep replaced with /.
+  const normalizedManifest = filteredFileDetails.map((fileDetails) => {
     return {
-      url: url,
+      url: fileDetails.file.replace(path.sep, '/'),
       revision: fileDetails.hash,
     };
   });
+
+  const manifestTransforms = [];
+
+  if (options.modifyUrlPrefix) {
+    manifestTransforms.push(modifyUrlPrefixTranform(options.modifyUrlPrefix));
+  }
+
+  if (options.dontCacheBustUrlsMatching) {
+    manifestTransforms.push(
+      noRevisionForUrlsMatchingTransform(options.dontCacheBustUrlsMatching));
+  }
+
+  if (Array.isArray(options.manifestTransforms)) {
+    manifestTransforms.concat(options.manifestTransforms);
+  }
```

Comment:
Is it worth throwing an error if it's not an array but is defined? Originally I was expecting this to be "manifestTransform" and the use of this would be a single function (i.e. it would be up to the developer to manage multiple transforms etc. Fine with offering the API as an array, but I can expect people to set it to a function and then wonder why it's never called.